common/mariadb/templates/pre-change-job.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if or .Values.databases .Values.users }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ include "fullName" . }}-pre-change"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ default "unknown" .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    "helm.sh/hook": pre-upgrade,pre-rollback
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{ include "fullName" . }}-pre-change"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ .Values.name }}-db-exec
+      containers:
+      - name: pre-change-job
+        image: "{{ required ".Values.global.dockerHubMirrorAlternateRegion is missing" .Values.global.dockerHubMirrorAlternateRegion }}/{{ .Values.pre_change_hook.image }}:{{ .Values.pre_change_hook.image_version }}"
+        command:
+        - kubectl-v{{ .Values.pre_change_hook.kubectl_version }}
```

Comment:
Isn't `kubectl` in the image  wrapper that automatically uses the appropriate version? Or is this to guard against an old image being used and thus not having the version yet?

Reply:
The wrapper is for a different use-case ("out of cluster") and doesn't work with an in-cluster setup.

It requires a user and a password so it can do its magic (version discovery) and log-on to any kubernetes api endpoint and works with a kubeconfig.
The script relies on the in-cluster config and the credentials provided by the service-account.

Since we are now more strict on following upstream versions of kubernetes, I do not expect a large drift between versions across the pipeline, which requires an automatic version detection. So I think, we can avoid injecting the credentials and having an additional step (version discovery), which might fail. Still, it can be overwritten in each region, should the need arise.